tabrepo/contexts/context.py

```diff
@@ -275,78 +279,90 @@ def download(self,
                 Guarantees alignment between local and remote files (at the time of download)
         :param dry_run: If True, will not download files, but instead log what would have been downloaded.
         """
-        print(f'Downloading files for {self.name} context... '
-              f'(include_zs={include_zs}, exists="{exists}", dry_run={dry_run})')
-        if dry_run:
-            print(f'\tNOTE: `dry_run=True`! Files will not be downloaded.')
-        assert exists in ["raise", "ignore", "overwrite"]
-        assert self.s3_download_map is not None, \
-            f'self.s3_download_map is None: download functionality is disabled'
-        file_paths_expected = self.benchmark_paths.get_file_paths(include_zs=include_zs)
-
-        file_paths_to_download = [f for f in file_paths_expected if f in self.s3_download_map]
-        if len(file_paths_to_download) == 0:
-            print(f'WARNING: Matching file paths to download is 0! '
-                  f'`self.s3_download_map` probably has incorrect keys.')
-        file_paths_already_exist = [f for f in file_paths_to_download if self.benchmark_paths.exists(f)]
-        file_paths_missing = [f for f in file_paths_to_download if not self.benchmark_paths.exists(f)]
-
-        if exists == 'raise':
-            if file_paths_already_exist:
-                raise AssertionError(f'`exists="{exists}"`, '
-                                     f'and found {len(file_paths_already_exist)} files that already exist locally!\n'
-                                     f'\tExisting Files: {file_paths_already_exist}\n'
-                                     f'\tMissing  Files: {file_paths_missing}\n'
-                                     f'Either manually inspect and delete existing files, '
-                                     f'set `exists="ignore"` to keep your local files and only download missing files, '
-                                     f'or set `exists="overwrite"` to overwrite your existing local files.')
-        elif exists == 'ignore':
-            file_paths_to_download = file_paths_missing
-        elif exists == 'overwrite':
-            file_paths_to_download = file_paths_to_download
-        else:
-            raise ValueError(f'Invalid value for exists (`exists="{exists}"`). '
-                             f'Valid values: {["raise", "ignore", "overwrite"]}')
+        if verbose:
+            print(f'Downloading files for {self.name} context... '
+                  f'(include_zs={include_zs}, exists="{exists}")')
+        if use_s3:
```

Comment:
Can we move this into a `download_from_s3` method? That way it will look a good bit cleaner.

Reply:
Done.